gpii/node_modules/lifecycleManager/src/LifecycleManagerSession.js

```diff
@@ -72,17 +72,55 @@ var gpii = fluid.registerNamespace("gpii");
     });
 
     /**
-     * Save prefs to the cloud when the preferences are updated. Note that change requests on session.model.preferences
+     * Filter the preferences by metadata defined in the preferences set. Only the preferences that are allowed to be
+     * autosaved will be returned.
+     * @param {Object} preferences - The updated preferences.
+     * @param {Object} metadata - The metadata that defined in the prefs set.
+     * @return {Object} - The preferences to be autosaved.
+     */
+    gpii.lifecycleManager.session.filterPrefsByAutoSave = function (preferences, metadata) {
+        var autosaveScope;
+        fluid.each(metadata, function (meta) {
+            if (meta.type === "autosave") {
+                autosaveScope = fluid.makeArray(meta.scope);
+            }
+        });
+
+        // Preferences will not be auto saved if the "autosave" metadata is not defined
+        if (!autosaveScope) {
+            return undefined;
+        }
+
+        var prefsForAutoSave;
+        fluid.each(preferences, function (value, preference) {
```

Comment:
Note that as per the snapset2a shown, these may not be flat. One route around this might be to just write a nested loop when encountering applications, and accumulate a 2-element path which you then send to fluid.get in your isomorphic scope object as constructed above.